src/fides/api/models/privacy_notice.py

```diff
@@ -160,6 +162,15 @@ def applies_to_system(self, system: System) -> bool:
                     return True
         return False
 
+    @classmethod
+    def generate_notice_key(cls, name: Optional[str]) -> FidesKey:
+        """Generate a notice key from a notice name"""
+        if not isinstance(name, str):
+            raise Exception("Privacy notice keys must be generated from a string.")
+        notice_key: str = re.sub(r"\s+", "_", name.lower().strip())
+        FidesKey.validate(notice_key)
+        return notice_key
```

Comment:
Not yet being used, but I imagine I can do something like this when we start generating notices out of the box - or! we can already have the notice key in the yaml as well.

Reply:
We can go either way here! In the current form, I make `notice_key` required, but I can make it optional and use this instead.  That would also help separate the frontend from the backend.